opensearch-operator/pkg/reconcilers/ismpolicy.go

```diff
@@ -146,6 +146,7 @@ func (r *IsmPolicyReconciler) Reconcile() (retResult ctrl.Result, retErr error)
 	if err != nil {
 		reason := "error creating opensearch client"
 		r.recorder.Event(r.instance, "Warning", opensearchError, reason)
+		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, err
```

Comment:
The correct way here is to set `retResult` and `retErr` as the defer logic will use that to set a corresponding status in kubernetes. Similar to how the blocks above do it.

Reply:
Thanks for the feedback. I updated the PR